app/celery/letters_pdf_tasks.py

```diff
@@ -101,19 +102,27 @@ def get_pdf_for_templated_letter(self, notification_id):
 
 
 @notify_celery.task(bind=True, name="update-billable-units-for-letter", max_retries=15, default_retry_delay=300)
-def update_billable_units_for_letter(self, notification_id, page_count):
+def update_billable_units_or_validation_failed_for_templated_letter(self, notification_id, page_count,
+                                                                    create_fake_letter_response_file=None):
     notification = get_notification_by_id(notification_id, _raise=True)
-
-    billable_units = get_billable_units_for_letter_page_count(page_count)
-
-    if notification.key_type != KEY_TYPE_TEST:
-        notification.billable_units = billable_units
+    if page_count > LETTER_MAX_PAGE_COUNT:
+        notification.status = NOTIFICATION_VALIDATION_FAILED
         dao_update_notification(notification)
-
-        current_app.logger.info(
-            f"Letter notification id: {notification_id} reference {notification.reference}: "
-            f"billable units set to {billable_units}"
-        )
+        current_app.logger.info(f"Letter notification id: {notification_id} reference {notification.reference}: "
+                                f"validation failed: letter is too long {page_count}")
+    else:
+        if notification.key_type != KEY_TYPE_TEST:
+            notification.billable_units = get_billable_units_for_letter_page_count(page_count)
+            dao_update_notification(notification)
+            current_app.logger.info(f"Letter notification id: {notification_id} reference {notification.reference}: "
+                                    f"billable units set to {notification.billable_units}")
+        if notification.key_type == KEY_TYPE_TEST \
+                and current_app.config['NOTIFY_ENVIRONMENT'] in ['preview', 'development']:
+            # For test letters send a fake letter response
+            create_fake_letter_response_file.apply_async(
```

Comment:
True - however, it is possible to clients to send a test letter and it should show that the letter is too long. It might surprise a client that the letter fails for a live key and not a test key. I'll take another swing at this.

Reply:
yeah, i agree that test letters should deffo fail validation.

but it looks like this specific code block only deals with sending fake dvla ack files on preview and development though. it shouldn't affect anyone on prod

Reply:
In that case I'll put this back in post_notifications.